sbtc/tests/integration/validation.rs

```diff
@@ -386,3 +400,177 @@ fn op_csv_disabled() {
+    // Nice, the depositor spent all of their funds on the deposit.
+    assert_eq!(depositor.get_balance(rpc).to_sat(), 0);
+
+    // Let's check that this is a valid sBTC deposit.
+    let request = CreateDepositRequest {
+        outpoint: OutPoint::new(deposit_tx.compute_txid(), 0),
+        reclaim_script: reclaim_script.clone(),
+        deposit_script: deposit_script.clone(),
+    };
+
+    let _ = request.validate_tx(&deposit_tx).unwrap();
+
+    // Alright now we know the signers haven't moved our funds, so we
+    // reclaim it. We construct a transaction spending the funds back to
+    // ourselves.
+    let reclaim_tx_fee = 30000;
+    let mut reclaim_tx = Transaction {
+        version: Version::TWO,
+        lock_time: LockTime::from_height(5).unwrap(),
+        input: vec![TxIn {
+            previous_output: OutPoint::new(deposit_tx.compute_txid(), 0),
+            sequence: Sequence::from_consensus(reclaim.lock_time()),
+            script_sig: ScriptBuf::new(),
+            witness: Witness::new(),
+        }],
+        output: vec![TxOut {
+            value: Amount::from_sat(amount_sats - reclaim_tx_fee),
+            script_pubkey: depositor.script_pubkey.clone(),
+        }],
+    };
+
+    // Now we need to sign our input. We construct the sighash and create a
+    // BIP-341 Schnorr signature. This part is not obvious.
+    let input_utxos: Vec<TxOut> = vec![deposit_utxo];
+
+    let prevouts = Prevouts::All(input_utxos.as_slice());
+    let sighash_type = TapSighashType::Default;
+    let mut sighasher = SighashCache::new(&reclaim_tx);
+
+    let leaf_hash = TapLeafHash::from_script(&reclaim_script, LeafVersion::TapScript);
+    let reclaim_sighash = sighasher
+        .taproot_script_spend_signature_hash(0, &prevouts, leaf_hash, sighash_type)
+        .unwrap();
+    let msg = secp256k1::Message::from(reclaim_sighash);
+
+    // Now we sign the sighash
+    let signature = SECP256K1.sign_schnorr(&msg, &depositor.keypair);
+
+    // Now we need to construct the merkle path proving that the merkle
+    // tree is what we say it is.
+    let ver = LeafVersion::TapScript;
+    // For such a simple tree, we construct it by hand.
+    let leaf1 = NodeInfo::new_leaf_with_ver(deposit_script.clone(), ver);
+    let leaf2 = NodeInfo::new_leaf_with_ver(reclaim_script.clone(), ver);
+
+    let node = NodeInfo::combine(leaf1, leaf2).unwrap();
+    let internal_key = *sbtc::UNSPENDABLE_TAPROOT_KEY;
+
+    let taproot = TaprootSpendInfo::from_node_info(SECP256K1, internal_key, node);
+    let control_block = taproot
+        .control_block(&(reclaim_script.clone(), ver))
+        .unwrap();
+    // Now we can set the witness data.
+    let witness_data = [
+        signature.serialize().to_vec(),
+        reclaim_script.to_bytes(),
+        control_block.serialize(),
+    ];
+    reclaim_tx.input[0].witness = Witness::from_slice(&witness_data);
+
+    // Not enough time has passed so the submitting the reclaim transaction
```

Comment:
I think that we should do a `faucet.generate_blocks(lock_time as u64 - 1)` here, and change the `generate_blocks()` below to `1`, just to show that things work as expected around the boundary.

Reply:
Great point. Doing so highlights an off-by-one error. Maybe we should increase out buffer by one?

Reply:
Well not an error, but a way of thinking. You'll see